This adds the maturity model as proposed in #116

MaineC opened the pull request:
This pull requests converts the maturity model proposed in #116 into pattern
format. It converts the table shared in the issue into something that hopefully
is easier to review - but also easier to reference. I followed the format of the
Apache Maturity Model
https://community.apache.org/apache-way/apache-project-maturity-model.html for
formatting the individual points. Links (anchors in the page to individual
points) might make referencing points more precise but are missing at this
point.

I converted the table as is - we can tidy up and clarify going forward. At least
IMHO there are still a few points that are debatable.

MaineC committed:
This adds the maturity model as proposed in #116

This pull requests converts the maturity model proposed in #116 into pattern
format. It converts the table shared in the issue into something that hopefully
is easier to review - but also easier to reference. I followed the format of the
Apache Maturity Model
https://community.apache.org/apache-way/apache-project-maturity-model.html for
formatting the individual points. Links (anchors in the page to individual
points) might make referencing points more precise but are missing at this
point.

I converted the table as is - we can tidy up and clarify going forward. At least
IMHO there are still a few points that are debatable.

MaineC commented on maturity-model.md:
```diff
@@ -0,0 +1,175 @@
+
+### Transparency
+
+**Plans & Products**
```
I like how this proposal originally coming from @dicortazar takes planning into consideration as well.

MaineC commented on maturity-model.md:
```diff
@@ -0,0 +1,175 @@
+**Plans & Products**
+
+* PP-0: Individuals and teams do not regularly disclose their plans or products to multiple stakeholders. No formal actions exists at the organization.
+* PP-1: Individuals and teams give visibility to their plans or products to multiple stakeholders, before they are started. Shared roadmap.
```
This mentions that there is a shared roadmap. In the two points below we do not talk about whether or not the roadmap should remain shared - also, is that "roadmap for this InnerSource project is shared with others" or is it "the organisation has one shared roadmap for all projects". I would assume the former but maybe making that explicit would make it clearer.

dicortazar replied:
Thanks @MaineC for your comments.
Do you think it makes sense to state explicitly in each of the following two bullet points that the roadmap is already shared?

A proposal:
PP-2: There are already shared roadmaps with clear guidelines and contribution rules where stakeholders can provide feedback. However this is not standardized across the organization and not all of the projects provide this info.
PP-3: Roadmaps are shared by default and there is a standard process and homogeneous way to do this across the organization at the level of each InnerSource project. This contains clear rules to contribute and influence in the roadmap.

MaineC replied:
I like your proposed re-formulation of PP-2 and PP-3. They also make it clear that we are talking about separate roadmaps per project - each of which is shared.

MaineC commented on maturity-model.md:
```diff
@@ -0,0 +1,175 @@
+
+
+**Development Process & Tools**
+
```
I believe we are only scratching the surface here - this focusses on CI, code repositories and code review. Maybe rename the point to something more specific? Though I'm not sure to what.

MaineC replied:
Also what I'm missing in this point is a mention of how InnerSource projects can define their own development process, but should publish that in a Contributing.md file. I believe this level of autonomy coupled with the transparency of having to communicate this process is vital when scaling InnerSource as it allows teams to develop independently.

Another point I'm missing is that in mature InnerSource projects I would expect process changes to happen through Pull Requests to those documents containing the process description: You want us to do code review differently? Post a pull request changing our contributing.md file and we'll take it from there. That way even governance can happen in an async fashion.

dicortazar replied:
Regarding to your second comment, I'd say that it is good that each InnerSource project defines its own process, but at the same time, one of the goals of InnerSource is to create homogeneous ways of contributing across the organization. Thus, having a similar infra (e.g., GitHub/GitLab/etc) may help in this process.

Thus, as an overall goal here: the purpose of this section was to provide a way to assess the development process (mainly taking into account the first areas you mention such as CI, repositories, code review), but having under control the myriad of options out there and defining a common process within the organization.

At the InnerSource Commons we saw some cases where organizations are facing the issue of having isolated technological islands (e.g., certain programming language, certain tooling, certain process, etc). But when this is so specific to the team idiosyncrasy, and this team does not exist any more, then we have the maintenance issues, etc.

So, the proposal here is more focused on having a mature development process, but at the same time creating an homogeneous way of doing this across the organization.

What do you think @MaineC ?

MaineC replied:
After thinking back and forth and comparing to open source projects, I agree with the point you are making, but would phrase it differently. .oO(this will take some typing to expand)

Regarding defining your own processes: I still think there is value in that. Think "how many trusted committer approvals do we require before merging?", "how should you approach us when making changes - do we always require an issue up front, or are we ok with receiving a pull request right away?", "exactly which issues will you run into when running our test suite?"

I believe giving projects some flexibility here helps reduce up-front coordination on how teams want to operate. (Think about the never ending discussion of where the curly braces should be put - over time, I have stopped worrying about that and simply follow the style that is already being used in the project.)

On the other end of the spectrum there are projects that move to exotic version control systems and build their own build systems. This happens in the open source world as well - with different degrees of success: Chrome (IIRC) at some point had to move to a more standard build tool to gain more contributors. The Linux kernel managed to establish their own version control system as the new normal.

In mature Inner Source projects me personally, I would expect trusted committers to be able to make a call which technology is best for the project - balanced with which technology is best to increase the likelihood of finding new contributors (use standard tooling, if you want to boost adoption and contributions).

In less mature organisations I would expect either everything being prescribed - or decisions on which to use being based not with the goal of increasing contribution levels, but based on team internal metrics only.

Right now, I have no idea how to transform this line of thinking into clear maturity levels - not even sure if that would help those adopting InnerSource who do not have years of Open Source experience - including years of searching for new contributors. It takes a while to realize that it takes people to write software.

MaineC commented on maturity-model.md:
```diff
@@ -0,0 +1,175 @@
+* DP-0:	Each team follows its own development process and tools. They are not defined to share knowledge and artifacts outside development team. Siloed development teams.
+* DP-1: Development teams use shared code repositories, internally. Some teams develop their own CI process, using non corporate or standard CI tools. There is no code review process defined, although some projects teams do it internally.
+* DP-2: The organization sponsors and promotes a shared repository for collective knowledge. Some teams develop their own CI process, using corporate CI tools. There are CI environments. Code review process defined, and used by some projects. Code review is sometimes done by external team members.
+* DP-3: Most teams develop their own CI process, using corporate CI tools. There are CI environments. Code review process defined, and used by some projects. Code review is usually done by external team members.
```
I don't understand the point about code review usually being done by external team members.

dicortazar replied:
I think we can simplify this to the fact that there is a proper collaboration in place. If we reach DP-3, this would mean that code review is already in place, and people and different business units are reviewing each other.

MaineC commented on maturity-model.md:
```diff
@@ -0,0 +1,175 @@
+
+**Decisions**
+
+* DC-0: Decision-makers often withhold data and resources without explanation.
```
```suggestion
* DC-0: Decision-makers often intentionally or accidentally withhold data and resources related to project decisions.
```

MaineC commented on maturity-model.md:
```diff
@@ -0,0 +1,175 @@
+* DP-2: The organization sponsors and promotes a shared repository for collective knowledge. Some teams develop their own CI process, using corporate CI tools. There are CI environments. Code review process defined, and used by some projects. Code review is sometimes done by external team members.
+* DP-3: Most teams develop their own CI process, using corporate CI tools. There are CI environments. Code review process defined, and used by some projects. Code review is usually done by external team members.
+
+**Decisions**
```
I wonder if one should distinguish company wide decisions and InnerSource project internal decisions?

MaineC commented on maturity-model.md:
```diff
@@ -0,0 +1,175 @@
+* RS-2: Individuals and teams make project-related materials accessible to some members of project teams according to clearly defined and shared formats and/or protocols. Individuals and teams withhold sensitive data and resources, provide limited details, context, and scope.
+* RS-3: Individuals and teams make project-related materials broadly accessible to the organization—and possibly outside the organization as well—according to clearly defined and shared formats and/or protocols. Individuals and teams who must withold sensitive data and resources are clear about what they're not sharing, and others understand why those materials are not available to them.
+
+**Stories**
```
While this is important for InnerSource to thrive I wonder if this could be phrased more specifically for InnerSource or at least be extended by an explanation of why that is important? Thinking about it, the "why is this point important" likely should be added to all of these points.

dicortazar replied:
Perhaps experiences?

The rationale for this point is that it is important to learn from each others, no matters if these are failures or successes.

MaineC replied:
I tried to re-phrase and add as a rationale. Feedback welcome.

MaineC commented on maturity-model.md:
```diff
@@ -0,0 +1,175 @@
+### Collaboration
+
+**Channels for Providing Feedback**
+
```
This point seems to mix several aspects into one.

Likely it tries to talk about "Communication channels for projects", "Feedback culture", "Channels for development discussions", "Channels for user support"?

dicortazar replied:
Initially any channel of communication that helps to start building a community, or several of them, within the organization.

Thus, there final goal is to have proper communication channels known by anyone in the organization and potentially with different goals (announcements, training, project channels, infra discussions, etc).

The goal here is not to define the really specific communication channels you need, but make it clear that you need them and that they should follow certain guidelines (e.g., netiquette). This again helps to have an homogeneous way of behaving within the organization (e.g., using a specific provider for this), following certain rules (e.g., create a project channel each time there's a new project, open the channel to the public, foster discussions in the channels), and with the goal of having a referenceable and archivable body of knowledge (e.g., as in the case of the ASF with mailing list discussions).

MaineC replied:
Likely this goal should then be added as further explanation of the point :)

MaineC replied:
I tried to rephrase and add as an introduction sentence.

MaineC commented on maturity-model.md:
```diff
@@ -0,0 +1,175 @@
+
+* SP-0: No sharing culture nor written policies.
+* SP-1:	Some members of the organization unite to define values and principles, but are not clearly supported when they do.
+* SP-2:	Members of the organization collectively document shared visions and agreements like mission statements and codes of conduct, make them easily accessible, and reference them often. Onboarding materials and orientation rituals provide adequate context for helping new members understand how the organization will benefit from their contributions.
```
This sounds more like "how mature is the entire organisation to adopt InnerSource" than "how mature is the InnerSource project itself". The deeper I dive, I think these two questions should be separated. In some of the points above we seem to be speaking about the setup of individual InnerSource projects. In others about the entire organisation.

dicortazar replied:
The goal here is to have a common way of behaving across the organization. And for this, there is a need to build certain set of rules and guidelines that everyone agrees, knows and is able to reference if needed at some point.

When discussing about InnerSource in the previous sections, the goal at the very end is to have a common set of rules and guidelines we can all discuss about, share and reference if needed. It happens that depending on the level, it may make sense to start at the project level or at the organization level.

I see for instance that if we have a new tool, let's say Slack/Mattermost/RocketChat/etc.. to communicate internally, this should/may be fostered by the organization or the InnerSource program within the organization. And this is another tool that will hopefully make more productive/collaborative/communicative the several developers and contributors.

On the other hand, there might be cases where at the level of project, a specific group of people or the trusted committer decides that when coming to my project in my internal infrastructure, you should behave in a certain way and that the code review process takes place in a certain way due to project constraints. These constraints are project related and not organization related, while certain behaviors make more sense to have them discussed at the organization level.

Summarizing: the approach is top down or down top depending on the case and the maturity level. We may specify this specific context in each of the maturity levels if you think this is useful.

An example of this would be to state (just an example) that communication is a more top down approach where the InnerSource initiative will provide the tooling, training, and certain guidelines (once the community start using that tool). And then the specific guidelines about how to contribute to a certain project, although narrowed by the specific infra used (Gerrit, PRs, MRs, etc), are decided by each of the InnerSource projects (e.g., having 2 +2 positive votes as done in OpenStack instead of only one +2 vote).

What do you think about this?

MaineC replied:
I think I understand now better where you are going with this point. I've tried to capture it in the motivation for that point in the model. Feedback welcome.

MaineC replied:
And thank you for taking the time to type up the explanation above.

MaineC commented on maturity-model.md:
```diff
@@ -0,0 +1,175 @@
+* SP-2:	Members of the organization collectively document shared visions and agreements like mission statements and codes of conduct, make them easily accessible, and reference them often. Onboarding materials and orientation rituals provide adequate context for helping new members understand how the organization will benefit from their contributions.
+* SP-3:	Shared values and principles inform decision-making, conflict resolution, and assessment processes among members of the organization, who reference these values and principles consistently in both verbal and written formats.
+
+**Feel part of the Organization**
```
The points below sound fairly vague and like they apply to any organisation - not only to those trying to adopt InnerSource. I'd likely remove them or try to make them more relevant to the InnerSource context.

dicortazar replied:
Having a sense of ownership to a certain project, and increasing the engagement within the organization is one of the points of having an InnerSource initiative in place.

We tried to use this concept of 'feeling part of something', but happy to reword this to really include the concept of engagement. Any suggestion?

MaineC replied:
Hmm - you explaining that increasing engagement can be one reason for introducing an InnerSource initiative helped me understand better, why this is part of the capabilities model. Do you think adding this explanation would make sense?

MaineC commented on maturity-model.md:
```diff
@@ -0,0 +1,175 @@
+* RW-3:	Rewards are not only proposed by the organization, but the communities are able to define their more valuable rewards. The community is responsible to decide who has to be rewarded.
+
+**Monitoring Policies**
+
```
While important, this needs to be formulated in a much more InnerSource specific way to make sense in that context. Also it could be interpreted two ways:

InnerSource projects using metrics internally to track deployment performance, to track project progress.

InnerSource program metrics to track InnerSource adoption.

All of these are important, but I believe we should make explicit which we are talking about here.

dicortazar replied:
I'd say this should include both cases, projects self-assessing and to monitor the InnerSource initiative as a whole.

What about the following?

MP-0: No existing monitoring policies at any level in the organization.
MP-1:  Metrics are important for certain teams, and they start using them in an isolated way.
MP-2: There is a strategy at the organizational level with respect to metrics that help to validate specific policies across the organization. This monitoring policy exists at the level of some of the InnerSource projects.
MP-3: There are clear guidelines, recommendations, and trainings about the use of metrics with certain infrastructure provided by the organization. This works at both levels: InnerSource program to understand the general InnerSource adoption within the organization, and at the level of InnerSource projects.

MaineC replied:
+1, I've changed the wording as per your suggestion and added some additional rationale to the section.

MaineC commented on maturity-model.md:
```diff
@@ -0,0 +1,175 @@
+* MP-2:	Existing process at the organization.
+* MP-3:	Clear rules and processes in the use of metrics with specific training and monitoring in place.
+
+**Support and Maintenance**
```
Again this sounds more like a point that is relevant for pretty much any tech department and not InnerSource specific.

To make it InnerSource specific we could talk about adoption of the 30-day warranty and service vs. library patterns.

dicortazar replied:
That's a good idea: link to existing patterns or to the creation of new patterns to open certain discussions.

I think it makes sense to have this across the model.

MaineC commented on maturity-model.md:
```diff
@@ -0,0 +1,175 @@
+* SP-3:	There are rules and regulations to formalize the support on the product, given by a mature community.
+
+**Culture**
+
```
I added some prose to each point below, please correct where I added the wrong prose.

dicortazar replied:
+1 :)

MaineC commented on maturity-model.md:
```diff
@@ -0,0 +1,175 @@
+* CL-3: Activist - teams actively seek help, mentor and recruit new contributors.
+
+**InnerSource Roles**
+
```
Again I would split this into organisation vs. specific project.

MaineC reviewed (commented):
Thanks for writing the initial draft @dicortazar - I added some comments to get the discussion started below.

MaineC committed:
Re-formulate roadmap sharing aspect.

MaineC committed:
Clarify who is encouraged to do code review

MaineC committed:
Commit suggestion on silo

MaineC committed:
Add reasons for each model step

MaineC committed:
Expand goals with communication channel point.

MaineC committed:
Make need for sharing principles explicit.

MaineC committed:
Add rationale and more prose to metrics section

MaineC committed:
Added support patterns to respective level

MaineC committed:
Adjust maturity model to new structure

... plus a few minor formatting fixes in the pattern itself. Also added
co-authors from the original issue.

spier commented on patterns/2-structured/maturity-model.md:
```diff
@@ -0,0 +1,231 @@
+* PP-0: Individuals and teams do not regularly disclose their plans or products to multiple stakeholders. No formal actions exists at the organization.
+* PP-1: Individuals and teams give visibility to their plans or products to multiple stakeholders, before they are started. Shared roadmap.
+PP-2: There are already shared roadmaps with clear guidelines and contribution rules where stakeholders can provide feedback. However this is not standardized across the organization and not all of the projects provide this info.
+PP-3: Roadmaps are shared by default and there is a standard process and homogeneous way to do this across the organization at the level of each InnerSource project. This contains clear rules to contribute and influence in the roadmap.
```
Adding bullet points:

```suggestion
* PP-2: There are already shared roadmaps with clear guidelines and contribution rules where stakeholders can provide feedback. However this is not standardized across the organization and not all of the projects provide this info.
* PP-3: Roadmaps are shared by default and there is a standard process and homogeneous way to do this across the organization at the level of each InnerSource project. This contains clear rules to contribute and influence in the roadmap.
```

spier commented on patterns/2-structured/maturity-model.md:
```diff
@@ -0,0 +1,231 @@
+## Title
+Maturity model: Learn about InnerSource best practices
```
Adding spacing.

```suggestion
## Title

Maturity model: Learn about InnerSource best practices
```

spier commented on patterns/2-structured/maturity-model.md:
```diff
@@ -0,0 +1,231 @@
+
+## Problem
+
+When InnerSource adoption in an enterprise starts to increase individual
```
```suggestion
When InnerSource adoption in an enterprise starts to increase, individual
```

spier commented on patterns/2-structured/maturity-model.md:
```diff
@@ -0,0 +1,231 @@
+
+InnerSource projects thrive when contributors become active and participate. As a result, making contribution easier should be balanced with pure technical goals.
+
+* DP-0:	Each team follows its own development process and tools. They are not defined to share knowledge and artifacts outside development team. Siloed development teams.
```
```suggestion
* DP-0: Each team follows its own development process and tools. They are not defined to share knowledge and artifacts outside development team. Siloed development teams.
```

spier commented on patterns/2-structured/maturity-model.md:
```diff
@@ -0,0 +1,231 @@
+
+* DC-0: Decision-makers often intentionally or accidentally withhold data and resources related to project decisions.
+* DC-1: Materials that are part of decision-making practices become available for review after decisions are finalized.
+* DC-2:	People feel like they know about—and are helping to shape—most (but not all) important decisions as those decisions are unfolding. Materials that are part of decision-making practices are available at defined project milestones
```
```suggestion
* DC-2: People feel like they know about—and are helping to shape—most (but not all) important decisions as those decisions are unfolding. Materials that are part of decision-making practices are available at defined project milestones
```

spier commented on patterns/2-structured/maturity-model.md:
```diff
@@ -0,0 +1,231 @@
+In order to attract contributors helpful supporing material needs to be easily accessible.
+
+* RS-0: Individuals and teams neither contribute to nor draw upon a shared repository of knowledge.
+* RS-1:	Individuals and teams release project materials for review internally, after they've finished their work. Individuals and teams share resources, but in disconnected, fragmented, or individualized/siloed systems or repositories. Individuals and teams share resources, but there is no commonly expressed or shared understanding of the criteria used to determine whether information is sensitiv ...
```
```suggestion
* RS-1: Individuals and teams release project materials for review internally, after they've finished their work. Individuals and teams share resources, but in disconnected, fragmented, or individualized/siloed systems or repositories. Individuals and teams share resources, but there is no commonly expressed or shared understanding of the criteria used to determine whether information is sensitive or not. Do not "share thinking on others".
```

spier commented on patterns/2-structured/maturity-model.md:
```diff
@@ -0,0 +1,231 @@
+
+* ST-0: Individuals and teams don't share neither successes nor failures for others to learn from.
+* ST-1: Individuals and teams are comfortable sharing stories about successes, but not about failures.
+* ST-23	Individuals and teams are comfortable sharing stories of successes and failures during retrospectives and reviews.
```
Did you mean step 2 here instead of 23?

```suggestion
* ST-2: Individuals and teams are comfortable sharing stories of successes and failures during retrospectives and reviews.
```

spier commented on patterns/2-structured/maturity-model.md:
```diff
@@ -0,0 +1,231 @@
+* ST-0: Individuals and teams don't share neither successes nor failures for others to learn from.
+* ST-1: Individuals and teams are comfortable sharing stories about successes, but not about failures.
+* ST-23	Individuals and teams are comfortable sharing stories of successes and failures during retrospectives and reviews.
+* ST-3:	Individuals and teams are comfortable sharing stories of successes and failures, and learn from failures according to formal protocols.
```
```suggestion
* ST-3: Individuals and teams are comfortable sharing stories of successes and failures, and learn from failures according to formal protocols.
```

spier commented on patterns/2-structured/maturity-model.md:
```diff
@@ -0,0 +1,231 @@
+
+## Known Instances
+
+Entelgy, Zylk, Bitergia
```
Hoping to make it more clear that these are 3 separate instances.
With that we might be able to move this pattern to the Validated (i.e. the highest pattern maturity) in a future PR.

```suggestion
* Entelgy
* Zylk
* Bitergia
```

spier commented on patterns/2-structured/maturity-model.md:
```diff
@@ -0,0 +1,231 @@
+
+Ideally you will end up with proper communication channels that are known by anyone in the organization - with channels focussed on different goals (announcements, user support, development channels, infra discussions, etc.). Some of the best practices you will establish as your InnerSource projects mature: Adoption of netiquette guidelines, opening a proven set of standard channels (which are be ...
+
+* CF-0:	There are no processes nor stablished channels. Some members of the organization share materials via private channels or discussions.
```
```suggestion
* CF-0: There are no processes nor stablished channels. Some members of the organization share materials via private channels or discussions.
```

spier commented on patterns/2-structured/maturity-model.md:
```diff
@@ -0,0 +1,231 @@
+
+* CF-0:	There are no processes nor stablished channels. Some members of the organization share materials via private channels or discussions.
+* CF-1: The organization is in the process of establishing internal guidelines and channels for encouraging diverse points of view about company/departmental decisions, so that anyone belonging to the organization can use them. Some members of the organization share decision-making materials unformally using unofficially platforms. Leaders maintain at least one clear and direct channel for organi ...
+* CF-2:	The organization has established internal guidelines and channels, and provides specific resources (training programs, access to content, etc.), for encouraging and soliciting diverse points of view on team or decisions.
```
```suggestion
* CF-2: The organization has established internal guidelines and channels, and provides specific resources (training programs, access to content, etc.), for encouraging and soliciting diverse points of view on team or decisions.
```

spier commented on patterns/2-structured/maturity-model.md:
```diff
@@ -0,0 +1,231 @@
+
+InnerSource projects encourage employees to contribute to projects outside of the direct influence of their direct line manager. This needs a culture of trust.
+
+* LS-0:	Command & control culture, within a highly hierarchical organization.
```
```suggestion
* LS-0: Command & control culture, within a highly hierarchical organization.
```

spier commented on patterns/2-structured/maturity-model.md:
```diff
@@ -0,0 +1,231 @@
+
+* LS-0:	Command & control culture, within a highly hierarchical organization.
+* LS-1: Some leaders are open to receiving feedback and creating an environment where people feel safe providing it.
+* LS-2:	Most leaders in the organization are open to receiving feedback and creating an environment where people feel safe providing it. Leaders show passivity about understanding whether all members feel empowered and enabled to share. Organization encourages leaders to actively seek voices not present in dialog out for inclusion.
```
```suggestion
* LS-2: Most leaders in the organization are open to receiving feedback and creating an environment where people feel safe providing it. Leaders show passivity about understanding whether all members feel empowered and enabled to share. Organization encourages leaders to actively seek voices not present in dialog out for inclusion.
```

spier commented on patterns/2-structured/maturity-model.md:
```diff
@@ -0,0 +1,231 @@
+* OF-0:	Working groups tend to be static in terms of membership and skill sets.
+* OF-1:	Cross-functional teams exist, but team roles are often unclear and governance structures are vague.
+* OF-2:	Cross-functional teams are common, and teams post their roles and goals publicly.
+* OF-3:	Cross-functional teams are common and make their activities known broadly to the organization; in turn, the organization promotes best practices for working together.
```
```suggestion
* OF-0: Working groups tend to be static in terms of membership and skill sets.
* OF-1: Cross-functional teams exist, but team roles are often unclear and governance structures are vague.
* OF-2: Cross-functional teams are common, and teams post their roles and goals publicly.
* OF-3: Cross-functional teams are common and make their activities known broadly to the organization; in turn, the organization promotes best practices for working together.
```

spier commented on patterns/2-structured/maturity-model.md:
```diff
@@ -0,0 +1,231 @@
+* CB-0:	Completly siloed, no collaboration outside the teams. Just some collaborations due to cross-functional teams.
+* CB-1:	Members of the organization and teams collaborate but frequently say it's "too difficult". Teams infrequently revisit the outcomes of their collaborations.
+* CB-2:	Members of the organization and teams actively seek opportunities to collaborate. Teams routinely discuss, revisit and debate the outcomes of their collaborative efforts, and make these outcomes available by default.
+* CB-3:	Members of the organization collaborate both internally and externally in ways that benefit all involved. Teams routinely discuss, revisit and debate the outcomes of their collaborative efforts, and share their learnings outside the organization, and make these outcomes externally available by default.
```
Spaces, and one typo fix.

```suggestion
* CB-0: Completely siloed, no collaboration outside the teams. Just some collaborations due to cross-functional teams.
* CB-1: Members of the organization and teams collaborate but frequently say it's "too difficult". Teams infrequently revisit the outcomes of their collaborations.
* CB-2: Members of the organization and teams actively seek opportunities to collaborate. Teams routinely discuss, revisit and debate the outcomes of their collaborative efforts, and make these outcomes available by default.
* CB-3: Members of the organization collaborate both internally and externally in ways that benefit all involved. Teams routinely discuss, revisit and debate the outcomes of their collaborative efforts, and share their learnings outside the organization, and make these outcomes externally available by default.
```

spier commented on patterns/2-structured/maturity-model.md:
```diff
@@ -0,0 +1,231 @@
+* SP-0: No sharing culture nor written policies.
+* SP-1:	Some members of the organization unite to define values and principles, but are not clearly supported when they do.
+* SP-2:	Members of the organization collectively document shared visions and agreements like mission statements and codes of conduct, make them easily accessible, and reference them often. Onboarding materials and orientation rituals provide adequate context for helping new members understand how the organization will benefit from their contributions.
+* SP-3:	Shared values and principles inform decision-making, conflict resolution, and assessment processes among members of the organization, who reference these values and principles consistently in both verbal and written formats.
```
```suggestion
* SP-1: Some members of the organization unite to define values and principles, but are not clearly supported when they do.
* SP-2: Members of the organization collectively document shared visions and agreements like mission statements and codes of conduct, make them easily accessible, and reference them often. Onboarding materials and orientation rituals provide adequate context for helping new members understand how the organization will benefit from their contributions.
* SP-3: Shared values and principles inform decision-making, conflict resolution, and assessment processes among members of the organization, who reference these values and principles consistently in both verbal and written formats.
```

spier commented on patterns/2-structured/maturity-model.md:
```diff
@@ -0,0 +1,231 @@
+* PA-0:	Low engagement, no collaboration and people do not feel comfortable sharing with others.
+* PA-1:	Members of the organization feel comfortable sharing their thoughts and opinions without fear of retribution, but only in familiar domains. People understand that the best ideas win, and leadership responsibilities accrue to people with histories of contribution and commitment.
+* PA-2:	Members of the organization feel comfortable sharing their thoughts and opinions without fear of retribution. Leaders demonstrate dedication to the organization's shared values.
+* PA-3:	The organization is proactive in telling members that it benefits from their contributions; as such, members demonstrate shared consciousness and empowered execution, and feel a sense of responsibility to the community. Leaders understand that they grow by helping others grow, and they mentor junior members of the organization.
```
```suggestion
* PA-0: Low engagement, no collaboration and people do not feel comfortable sharing with others.
* PA-1: Members of the organization feel comfortable sharing their thoughts and opinions without fear of retribution, but only in familiar domains. People understand that the best ideas win, and leadership responsibilities accrue to people with histories of contribution and commitment.
* PA-2: Members of the organization feel comfortable sharing their thoughts and opinions without fear of retribution. Leaders demonstrate dedication to the organization's shared values.
* PA-3: The organization is proactive in telling members that it benefits from their contributions; as such, members demonstrate shared consciousness and empowered execution, and feel a sense of responsibility to the community. Leaders understand that they grow by helping others grow, and they mentor junior members of the organization.
```

spier commented on patterns/2-structured/maturity-model.md:
```diff
@@ -0,0 +1,231 @@
+* RW-0:	No rewards.
+* RW-1:	Leaders are encouraged to reward exceptional collaborations, but there are no policies or processes established.
+* RW-2:	Standard processes are established to reward collaborations outside the developers' teams. Team leaders or boards decide who has to be rewarded.
+* RW-3:	Rewards are not only proposed by the organization, but the communities are able to define their more valuable rewards. The community is responsible to decide who has to be rewarded.
```
```suggestion
* RW-0: No rewards.
* RW-1: Leaders are encouraged to reward exceptional collaborations, but there are no policies or processes established.
* RW-2: Standard processes are established to reward collaborations outside the developers' teams. Team leaders or boards decide who has to be rewarded.
* RW-3: Rewards are not only proposed by the organization, but the communities are able to define their more valuable rewards. The community is responsible to decide who has to be rewarded.
```

spier commented on patterns/2-structured/maturity-model.md:
```diff
@@ -0,0 +1,231 @@
+* SP-0:	Support given by the core dev or support team. A business contract guaranties the support. There is no knowledge about the product outside the team.
+* SP-1:	There are rules and regulations to formalize the support on the product, given by a dedicated supporting team.
+* SP-2: Support for InnerSource contributions is formalized through InnerSource patterns like [30 day warranty](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/master/30-day-warranty.md) or [service vs. library](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/master/service-vs-library.md).
+* SP-3:	There are rules and regulations to formalize the support on the product, given by a mature community.
```
```suggestion
* SP-0: Support given by the core dev or support team. A business contract guaranties the support. There is no knowledge about the product outside the team.
* SP-1: There are rules and regulations to formalize the support on the product, given by a dedicated supporting team.
* SP-2: Support for InnerSource contributions is formalized through InnerSource patterns like [30 day warranty](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/master/30-day-warranty.md) or [service vs. library](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/master/service-vs-library.md).
* SP-3: There are rules and regulations to formalize the support on the product, given by a mature community.
```

spier commented on patterns/2-structured/maturity-model.md:
```diff
@@ -0,0 +1,231 @@
+
+* RO-0:	There are no specific roles helping InnerSource adoption. Only common development roles are present: developer, analyst, tester, etc.
+* RO-1: Occasionally some individuals and teams contribute to other projects. These are technical contributions, where the user/contributor role is seen. For some teams, it can be identified at least one member being a technical reference, who explains the development process to other development team members. He/she could be a candidate for covering the trusted commiter role.
+* RO-2:	An InnerSource Officer role is in charge for governance and support, including processes, etc. Identifies the education needs and ensures it is provided to the organization. Leads and mentors the organization in the engagement in IS projects. Is the first formal step in the way, defining the IS vision and roadmap. The organization has defined a trusted committer role, being a point of con ...
```
Hope that "energizing" is the correct grammar here?

```suggestion
* RO-0: There are no specific roles helping InnerSource adoption. Only common development roles are present: developer, analyst, tester, etc.
* RO-1: Occasionally some individuals and teams contribute to other projects. These are technical contributions, where the user/contributor role is seen. For some teams, it can be identified at least one member being a technical reference, who explains the development process to other development team members. He/she could be a candidate for covering the trusted commiter role.
* RO-2: An InnerSource Officer role is in charge for governance and support, including processes, etc. Identifies the education needs and ensures it is provided to the organization. Leads and mentors the organization in the engagement in IS projects. Is the first formal step in the way, defining the IS vision and roadmap. The organization has defined a trusted committer role, being a point of contact/reference not only for dev team members but also for external contributors. There is a standard process describing how to contribute to the community, contributor role is present. Data Scientist role is in charge of managing the traces of activity left by the InnerSource initiative, needed to measure the IS evolution. 	Trusted committer role will evolve to a more technical profile, and a community manager will be in charge of "energizing" the community, being his main responsibility to attract and retain new developers/users (contributors/community members).
```

lenucksi committed:
Apply suggestions from code review

Co-authored-by: Sebastian Spier <github@spier.hu>

spier commented on patterns/2-structured/maturity-model.md:
```diff
@@ -0,0 +1,231 @@
+
+Currently in status "Structured".
+
+
```
```suggestion
```

spier reviewed (changes requested):
I took one quick review pass at this pattern, mostly looking at the formatting.

I replaced a bunch of tabs with spaces, as that was inconsistent through the bulleted lists (could go either way, and no, I don't want to get into the tabs-vs-spaces conversation :))

Other than that the pattern looks great, especially the very detailed breakdown of the checklist-style **Solution** section.

Will review the content in my next review.

lenucksi committed:
Apply suggestions from code review

Co-authored-by: Sebastian Spier <github@spier.hu>

lenucksi reviewed (approved):
Thanks, according to our contributors handbook this definitely fits into 2-structured and we could start to discuss and enrich everything to further improve this with the entire community.

I'm happy to merge this here or take in more feedback from e.g. @dicortazar or @spier. Otherwise happy to take that as separate pull requests. In that case please reference this one.

spier commented on patterns/2-structured/maturity-model.md:
```diff
@@ -0,0 +1,232 @@
+
+* SP-0: Support given by the core dev or support team. A business contract guaranties the support. There is no knowledge about the product outside the team.
+* SP-1: There are rules and regulations to formalize the support on the product, given by a dedicated supporting team.
+* SP-2: Support for InnerSource contributions is formalized through InnerSource patterns like [30 day warranty](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/master/30-day-warranty.md) or [service vs. library](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/master/service-vs-library.md).
```
Files have moved to a new place:

```suggestion
* SP-2: Support for InnerSource contributions is formalized through InnerSource patterns like [30 day warranty](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/master/patterns/2-structured/30-day-warranty.md) or [service vs. library](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/master/patterns/2-structured/service-vs-library.md).
```

spier reviewed (approved):
Fixing a link. This PR can totally be merged already, great stuff!

lenucksi committed:
Update patterns/2-structured/maturity-model.md

Co-authored-by: Sebastian Spier <github@spier.hu>
